pkg/resmap/resmap.go

```diff
@@ -37,6 +37,26 @@ import (
 // ResMap is a map from ResId to Resource.
 type ResMap map[resource.ResId]*resource.Resource
 
+// FindByGVKN find the matched ResIds by Group/Version/Kind and Name
+func (m ResMap) FindByGVKN(inputId resource.ResId) []resource.ResId {
+	var result []resource.ResId
+	for id := range m {
+		if id.Name() == inputId.Name() && id.Gvk().String() == inputId.Gvk().String() {
+			result = append(result, id)
+		}
+	}
+	return result
+}
+
+// FindOneByGVKN find the matched resource by Group/Version/Kind and Name
+func (m ResMap) FindOneByGVKN(inputId resource.ResId) (*resource.Resource, bool) {
```

Comment:
This name suggests it will return the first match of many.

Perhaps: `DemandOneMatchForId`?